Main.java

```diff
@@ -1,588 +1,287 @@
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileOutputStream;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
-import java.io.Serializable;
+import java.io.*;
 import java.util.ArrayList;
 import java.util.Scanner;
 
-class Food implements Serializable
-{
-    int itemno;
+class Food implements Serializable {
+    int itemNo;
     int quantity;   
     float price;
     
-    Food(int itemno,int quantity)
-    {
-        this.itemno=itemno;
-        this.quantity=quantity;
-        switch(itemno)
-        {
-            case 1:price=quantity*50;
-                break;
-            case 2:price=quantity*60;
-                break;
-            case 3:price=quantity*70;
-                break;
-            case 4:price=quantity*30;
-                break;
+    Food(int itemNo, int quantity) {
+        this.itemNo = itemNo;
+        this.quantity = quantity;
+        this.price = calculatePrice(itemNo, quantity);
+    }
+
+    private float calculatePrice(int itemNo, int quantity) {
+        switch (itemNo) {
+            case 1: return quantity * 50;
+            case 2: return quantity * 60;
+            case 3: return quantity * 70;
+            case 4: return quantity * 30;
+            default: return 0;
         }
     }
 }
-class Singleroom implements Serializable
-{
+
+class SingleRoom implements Serializable {
     String name;
     String contact;
     String gender;   
-    ArrayList<Food> food =new ArrayList<>();
+    ArrayList<Food> foodOrders = new ArrayList<>();
 
-   
-    Singleroom()
-    {
-        this.name="";
-    }
-    Singleroom(String name,String contact,String gender)
-    {
-        this.name=name;
-        this.contact=contact;
-        this.gender=gender;
+    SingleRoom(String name, String contact, String gender) {
+        this.name = name;
+        this.contact = contact;
+        this.gender = gender;
     }
 }
-class Doubleroom extends Singleroom implements Serializable
-{ 
+
+class DoubleRoom extends SingleRoom {
     String name2;
     String contact2;
     String gender2;  
-    
-    Doubleroom()
-    {
-        this.name="";
-        this.name2="";
-    }
-    Doubleroom(String name,String contact,String gender,String name2,String contact2,String gender2)
-    {
-        this.name=name;
-        this.contact=contact;
-        this.gender=gender;
-        this.name2=name2;
-        this.contact2=contact2;
-        this.gender2=gender2;
+
+    DoubleRoom(String name, String contact, String gender, String name2, String contact2, String gender2) {
+        super(name, contact, gender);
+        this.name2 = name2;
+        this.contact2 = contact2;
+        this.gender2 = gender2;
     }
 }
-class NotAvailable extends Exception
-{
+
+class NotAvailable extends Exception {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/shouryaj98/Hotel-Management-Project-Java/pull/54#issuecomment-2437632510>#b99405</a><span> - 10/25/2024, 12:22 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The NotAvailable exception class lacks a constructor and a serialVersionUID
</div>
<div id="fix">
<b>Fix:</b> Add a constructor and serialVersionUID to the NotAvailable exception class
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -53,6 +53,10 @@ class DoubleRoom extends SingleRoom {
 
  class NotAvailable extends Exception {
 +    private static final long serialVersionUID = 1L;
 +
 +    public NotAvailable(String message) {
 +        super(message);
 +    }
 +
      @Override
      public String toString() {
          return "Not Available!";
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged